src/mlpack/methods/ann/augmented/tasks/copy_impl.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file copy_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of CopyTask class
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "copy.hpp"
+
+#include <iostream>
+
+using std::cerr;
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+CopyTask::CopyTask(int maxLength, int nRepeats) :
+    maxLength(maxLength),
+    nRepeats(nRepeats)
+{
+  // Just storing task-specific paramters.
+}
+
+template<typename ModelType>
+double CopyTask::Evaluate(ModelType& model) {
```

Comment:
This sounds like a good idea to me, because I've already ran into similar issue when trying to implement `SortTask` - it was a *huge* copy-paste from `CopyTask`, which failed to work nonetheless :cry:.
However, I would also propose to create one more header file with evaluation functions (sklearn-ish), like this:
```c++
#include <score.hpp>
auto score_fn = scorers::sequence_precision; // number of sequences model has got right.
// As an option:
// auto precision = scorers::bit_precision; 
// = number of dataset bits model has got right.

// ... CopyTask defined as in the previous comment

copyTask.Sample(predictor_train, responses_train);
copyTask.Sample(predictor_test, responses_test);
arma::mat responses_test_pred; // or whatever type is required;

model.Train(predictor_train,  responses_train, optimizer);
model.Predict(predictor_test,  responses_test_pred);
auto score = score_fn(responses_test, responses_test_pred);
```

IMHO, this will be even more in accord with the concern separation principle.

At the moment, I'm at an internship at our local Web-development company, though, so I hope to try out the ideas in this thread tonight or in the next few days.